libraries/ESP8266WebServer/examples/Graph/data/index.htm

```diff
@@ -0,0 +1,527 @@
+    <span class="left">.&nbsp;Sampling&nbsp;period:&nbsp;</span>
+    <input id="periodField" type="number" min="0" max="10000" value="1000" step="100" size="4" class="left" />
+    <span class="left">&nbsp;ms&nbsp;</span>
+    <input id="periodRange" type="range"  min="0" max="10000" value="1000" step="100" />
+    <button id="startPauseButton">Pause</button>
+    <button id="clearButton" disabled="true">Clear</button>
+  </div>
+  <div id="charts">
+    <div class="chart-container" style="position: relative; height:50%;"><canvas id="digital" style="background-color:#000000"></canvas></div>
+    <div class="chart-container" style="position: relative; height:25%;"><canvas id="analog"></canvas></div>
+    <div class="chart-container" style="position: relative; height:25%;"><canvas id="heap" style="background-color:#EEEEEE"></canvas>
+  </div>
+  <script>  
+
+  /////////////////////////////////////
+  // CREATE DIGITAL CHART
+  
+  // Remember all GPIO datasets
+  var allDigitalDatasets = [];  
+
+  var gpioColors = ['#D2FF71', '#59FFF4', '#DB399C', '#4245FF', '#FFB312', '#BB54E9', '#0BD400', '#42C2FF', '#FFFFFF', '#FF4242'];
+ 
+  var digitalChart = new Chart(document.getElementById('digital'), {
+    // The type of chart we want to create
+    type: 'scatter',      
+
+    // The data for our dataset
+    data: {
+      // 17 GPIO datasets will be added by a loop at init time
+      datasets: []
+    },
+
+    // Configuration options go here
+    options: {
+      maintainAspectRatio: false,
+      title: {
+        display: true,
+        text: 'Digital I/Os'
+      },
+      scales: {
+        xAxes: [{
+          id:"x-axis",
+          ticks: {
+            minRotation:0,
+            maxRotation:0
+          }
+        }],
+        // 17 GPIO scales will be added by a loop at init time
+        yAxes: []
+      },
+      tooltips: {
+        intersect: false
+      },
+      animation: {
+          duration: 0
+      },
+      hover: {
+          animationDuration: 0
+      }
+    }
+  });
+
+
+  /////////////////////////////////////
+  // CREATE ANALOG CHART
+  
+  var analogChart = new Chart(document.getElementById('analog'), {
+    // The type of chart we want to create
+    type: 'scatter',
+
+    // The data for our dataset
+    data: {
+      datasets: [{
+        label: 'Analog',
+        backgroundColor: 'rgb(0,0,0,0)',
+        borderColor: 'rgb(255, 99, 132)',
+        data: [],
+        showLine:true
+      }]
+    },
+
+    // Configuration options go here
+    options: {
+      maintainAspectRatio: false,
+      title: {
+        display: true,
+        text: 'Analog input'
+      },
+      legend: {
+        display: false
+      },
+      scales: {
+        xAxes: [{
+          id:"x-axis",
+          ticks: {
+            minRotation:0,
+            maxRotation:0
+          }
+        }],
+        yAxes: [{
+          ticks: {
+            beginAtZero:true
```

Comment:
Maybe add 'min:0'  to this one since the analogInput can't be < 0?

Reply:
Although I cannot reproduce the issue of negative scale upon first load (FFx 76.0.1 (64 bits) Win10), I can get negative numbers using the "unzoom" feature (e.g. using mouse wheel).
Unfortunately, min:0 does not seem to prevent negative values from being shown :-(.
I guess the zoom plugin overrides the values set for the Y axis.
A single click on the graph should reset the scale, though...
I proposed the zoom plugin following a discussion with @devyte to allow both a "full range" view as well as a focus on small changes, because each has use cases, but it's not perfect :-/.